### PR TITLE
osltoy - Feature Request: custom #include search paths

### PR DESCRIPTION
**Pull request by maxwelliverson**
## Description

For Dev Days

#1862 

Addressing a osltoy feature request, I've added the ability to specify custom search paths. 
The changes consist of the following:

- Added a command line option `-I <path>` to specify search paths (`OIIO::ArgParse` unfortunately can't mirror `OSLCompiler`'s `-Ipath` syntax)
- OSLToyMainWindow now stores an array of `OSLCompiler` options alongside a flag indicating whether or not they should be regenerated before compilation. The only option currently able to be specified is `-Ipath`, but this should ease any future extension.
- Added a GUI component consiting of a window with a mutable list of the current set of search paths. This list is initially populated with any paths passed via the command line, but any modifications to the list will be reflected when the shader is next recompiled. Modifying this list does not currently force recompilation.
- Added an entry to the Tools menu to open the new GUI component.
- Added a line of output to the error window indicating what options the compiler was invoked with. I've defaulted to having it print for every run, but this may want to only print on failure.

## Tests

I did not add any automated tests for this feature, as is consistent with the rest of osltoy. I did, however, do some manual testing to make sure the functionality works as expected.

Shaders that failed to `#include` some file instead succeeded when the parent directory of that file was added to the list of search paths via both the command line and the GUI component. Respectively, when said search path is then removed via
the GUI component, any subsequent compilation attempts fail again. The GUI component otherwise behaves as expected.

## Checklist:

- [x] I have read the [contribution guidelines](../CONTRIBUTING.md).
- [x] I have updated the documentation, if applicable.
- [x] I have ensured that the change is tested somewhere in the testsuite (adding new test cases if necessary).
- [x] My code follows the prevailing code style of this project. If I haven't
  already run clang-format v17 before submitting, I definitely will look at
  the CI test that runs clang-format and fix anything that it highlights as
  being nonconforming.

### CODE REVIEW

File: src/osltoy/osltoyapp.cpp [hunk 1]
```diff
@@ -351,6 +352,276 @@ class OSLToyRenderView final : public QLabel {
...
+
+        m_maxIndexWithContent
+            = (int)paths.size()
+              - 1;  // ok that this is -1 if the paths are empty
+
+
+
+        auto initialLineCount = required_lines();
+
+        m_lines.reserve(initialLineCount);
```
Comment (lgritz) [inline src/osltoy/osltoyapp.cpp:471]: As a style nit, we tend to only put blank lines within a function body where the extra whitespace is clarifying -- for example, if you have several code statements that are doing a well defined task A, followed immediately by several statements that do a separate task B, then a single blank line between them can help understanding of the conceptual grouping of the lines into tasks. But if the space doesn't especially clarify the algorithm, you may as well omit it and benefit from the function being compact. Also, even when whitespace makes it more clear, a single line suffices.

Comment (maxwelliverson) [reply src/osltoy/osltoyapp.cpp:471]: Gotcha! That makes perfect sense; I definitely have a bad habit of inserting extra line breaks while working on something for my own legibility and subsequently forgetting to remove them later on. I'll fix that now.

**Comment by maxwelliverson** [comment]
Noting that when the automated test suite ran on Friday evening for this pull request, the icx build got caught in an infinite loop at some point while building the BVH. Not that it's particularly relevant to this, but I wanted to point it out as it appears to be another case of #1874 cropping up

**Comment by lgritz** [comment]
The icx thing is a known problem, we're looking into it

**Pull request merged**